File: Rust/fui/patches/windowing_qt/src/platform/qt/qt_wrapper/cpp/qclipboard.cpp
#include <QClipboard>
#include "qclipboard.h"

void QClipboard_setText(void *self, void *text, int mode)
{
    QClipboard *clipboard = static_cast<QClipboard *>(self);
	const QString *qtext = static_cast<const QString *>(text);
	clipboard->setText(*qtext, (QClipboard::Mode)mode);
}

void* QClipboard_text(void *self, int mode)
{
    QClipboard *clipboard = static_cast<QClipboard *>(self);
	QString result = clipboard->text((QClipboard::Mode)mode);
	if (result.isNull()) {
        return nullptr;
    }
    return static_cast<void *>(new QString(std::move(result)));
}

File: Rust/fui/patches/windowing_qt/src/platform/qt/qt_wrapper/qclipboard.rs
use super::QString;

pub struct QClipboard {
    pub this: *mut ::std::os::raw::c_void,
}

impl QClipboard {
    pub fn set_text(&self, text: &QString, mode: i32) {
        unsafe {
            crate::platform::qt::qt_wrapper::QClipboard_setText(self.this, text.this, mode);
        }
    }

    pub fn text(&self, mode: i32) -> QString {
        unsafe {
            let qstring_ptr = crate::platform::qt::qt_wrapper::QClipboard_text(self.this, mode);
            QString {
                this: qstring_ptr,
                is_owned: true,
            }
        }
    }
}
